src/ogr/geometry.jl

```diff
@@ -133,6 +133,15 @@ creategeom(geomtype::OGRwkbGeometryType)::IGeometry =
 unsafe_creategeom(geomtype::OGRwkbGeometryType)::Geometry =
     Geometry(GDAL.ogr_g_creategeometry(geomtype))
 
+# When the geometry type `T` is known, pass it wrapped in `Val` for type 
+# stability. `T` is usually equal to `geomtype`, except in the case 
+# of `geomtype == wkbLinearRing`, in which case `T` is `wkbLineString`
```

Comment:
So this single exception precludes using geomtype directly, also at `creategeom` at line [130](https://github.com/yeesian/ArchGDAL.jl/pull/316/files#diff-99ff903910c45c3a29f4c99d07cf05347c19c4f9f1119ec5485de8c0b0d39cacR130)?

Reply:
Yeah. We need to pass both `T` and `geomtype` as in this case they are not the same. GDAL takes a `wkbLinearRing` and returns a `kwbLineString`. Pretty annoying really.